test/view_component/integration_test.rb

```diff
@@ -214,12 +215,28 @@ class IntegrationTest < ActionDispatch::IntegrationTest
     assert_select "title", "Component Previews"
   end
 
+  test "test preview index renders custom application layout if configured" do
+    ViewComponent::Base.default_preview_layout = "admin"
+    get "/rails/view_components"
+    ViewComponent::Base.default_preview_layout = nil
```

Comment:
For clarity on the code under test it feels like we're getting to a place where this would be considerably clearer as its own test class with setup and teardown methods that manipulate `default_preview_layout`. As it's written I did a double take a) we repeat this pattern in many tests b) is it important to nil out `default_preview_layout` before asserting or could doing that have side effects on the asserted conditions?


@joelhawksley what are you thoughts on using RSepc? The ability to use nested describe blocks would be perfect for cases like this where you would use before/after blocks to do common setup and teardown tasks.

Reply:
No strong opinions here about the test suite, so I'll happily defer to the rest of y'all.

Reply:
IMHO RSpec would be a great addition to the development environment of the gem @jonspalmer

Reply:
How about we start by moving these tests to their own file?

I’d like to _try_ to stick with our current testing framework as it is consistent with Rails 😄

Reply:
@jonspalmer @joelhawksley If you need me to go rejigger the tests, just let me know

Reply:
In order not to disrupt the previous config I would suggest to set the value back instead of nil, we can go even a step further and create a helper method for this in the `test/test_helper.rb` file

```ruby
# test/test_helper.rb

...

def with_default_preview_layout(layout)
  old_value = ViewComponent::Base.default_preview_layout
  ViewComponent::Base.default_preview_layout = layout
  yield
  ViewComponent::Base.default_preview_layout = old_value
end
```
Then in our tests we can just use it like:
```ruby
test "test preview index renders custom application layout if configured" do
  with_default_preview_layout("admin") do
    get "/rails/view_components"
    assert_select "title", "ViewComponent - Admin - Test"
  end
end
```